gp_activetable.c

```diff
@@ -183,16 +429,43 @@ report_active_table_helper(const RelFileNodeBackend *relFileNode)
 	found = false;
 
 	MemSet(&item, 0, sizeof(DiskQuotaActiveTableFileEntry));
-	item.dbid = relFileNode->node.dbNode;
-	item.relfilenode = relFileNode->node.relNode;
-	item.tablespaceoid = relFileNode->node.spcNode;
+	item.dbid = relFileNode->dbNode;
+	item.relfilenode = relFileNode->relNode;
+	item.tablespaceoid = relFileNode->spcNode;
 
 	LWLockAcquire(diskquota_locks.active_table_lock, LW_EXCLUSIVE);
 	entry = hash_search(active_tables_map, &item, HASH_ENTER_NULL, &found);
-	if (entry && !found)
-		*entry = item;
+	
+	if (entry)
+	{
+		if (!found)
+		{
+			entry->dbid = item.dbid;
+			entry->relfilenode = item.relfilenode;
+			entry->tablespaceoid = item.tablespaceoid;
+		}
+		else
+		{
+			Oid relid;
+			LWLockAcquire(diskquota_locks.pg_class_cache_lock, LW_SHARED);
+			relid_cache_entry = hash_search(relid_cache, &relFileNode->relNode, HASH_FIND, &found);
+			if (found)
+			{
+				relid = relid_cache_entry->relid;
+			}
+			LWLockRelease(diskquota_locks.pg_class_cache_lock);
+
+			if (!found)
+			{
+				return;
+			}
 
-	if (!found && entry == NULL)
+			quota_check_common(relid_cache_entry->relid);
+			// TODO: check whether relation map already built
+			update_relation_map(relid_cache_entry->relid);
```

Comment:
```suggestion
			quota_check_common(relid);
			// TODO: check whether relation map already built
			update_relation_map(relid);
```

The lock `diskquota_locks.pg_class_cache_lock` is released here, hence it's not safe to use `relid_cache_entry`.

Reply:
The relid maybe INVALID, so we cannot use it to check quota limit.